lib/internal/Magento/Framework/View/Layout/etc/elements.xsd

```diff
@@ -138,7 +138,7 @@
 
     <xs:simpleType name="blockClassType">
         <xs:restriction base="xs:string">
-            <xs:pattern value="[A-Z][a-zA-Z\d]*(\\[A-Z][a-zA-Z\d]*)*"/>
+            <xs:pattern value="[A-Z][a-zA-Z\d]*(\\[A-Z][_a-zA-Z\d]*)*"/>
```

Comment:
For consistency with other places, like https://github.com/magento/magento2/pull/11650/files, were underscore was allowed previously, please allow underscore in vendor name also, perform an amend commit and force push into the same branch.

Reply:
I have made this change. Now both Vendor and Module allow underscores.

Reply:
Changes look good now. Just squash all changes into single commit and do a force push to the same branch please.

Reply:
I think I have done this but let me know if I've made a mistake or you need anything else.